src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,208 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private ListNode<T> first;
+    private ListNode<T> last;
+    private int size = 0;
+
+    private class ListNode<T> {
+        private T value;
+        private ListNode<T> prev;
+        private ListNode<T> next;
+
+        ListNode(T value) {
+            this.value = value;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
     @Override
     public void add(T value) {
+        ListNode<T> newNode = new ListNode<>(value);
+
+        if (first == null) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+        ListNode<T> newNode = new ListNode<>(value);
+
+        if (index == 0) {
+            if (size == 0) {
+                first = newNode;
+                last = newNode;
+            } else {
+                newNode.next = first;
+                first.prev = newNode;
+                first = newNode;
+            }
+        } else if (index == size) {
+            newNode.prev = last;
+            last.next = newNode;
+            last = newNode;
+        } else {
+            ListNode<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+
+        if (list == null || list.isEmpty()) {
+            return;
+        }
+
+        for (T elements : list) {
+            add(elements);
+        }
+
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        ListNode<T> current = first;
+
+        if (current == null) {
+            return null;
+        } else {
+            for (int i = 0; i < index; i++) {
+                if (current != null) {
+                    current = current.next;
+                }
+            }
+        }
+
+        if (current == null) {
+            return null;
```

Comment:
This null check is redundant because 'checkIndex' ensures the index is valid. If 'current' were null, it would indicate a logical error elsewhere.